### PR TITLE
docs: clarify the usage of gitLabAutomerge

### PR DESCRIPTION
**Pull request by kdev**
## Changes:

This PR clarifies the documentation

## Context:

See [this](https://github.com/renovatebot/renovate/discussions/10686)

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -676,6 +676,8 @@ Example:
 
 ## gitLabAutomerge
 
+Attention: This is a top-level key. To activate automerging for a specific `packageRule`, make sure to also set `automerge` to `true` in the specific rule!
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:679]:
```suggestion
Attention: This is a top-level key.
To activate automerging within a `packageRule`, you must set `automerge` to `true` in that same `packagaRule`.
```

We use one sentence per line pattern.

Maybe this rewrite of the second sentence is clearer on what you need to do?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 1 above)
Comment (kdev) [reply docs/usage/configuration-options.md:679]: Yeah, that sounds way better than mine!!

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -676,6 +676,9 @@ Example:
 
 ## gitLabAutomerge
 
+Attention: This is a top-level key.
+To activate automerging within a `packageRule`, you must set `automerge` to `true` in that same `packagaRule`.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:680]:
```suggestion
To activate automerging within a `packageRule`, you must set `automerge` to `true` in the same `packageRule`.
```

Whoops, made a spelling error.

And maybe `in the same` is better than `in that same`, but not sure about that one.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 2 above)
Comment (viceice) [reply docs/usage/configuration-options.md:680]: Automerge can be also globally used. Or you can use one of our automerge presets.

So this docs should describe that this gitlab flag only sets the gitlab flag when automerge is used too.

Comment (kdev) [reply docs/usage/configuration-options.md]: > 
> 
> Automerge can be also globally used. Or you can use one of our automerge presets.
> 
> So this docs should describe that this gitlab flag only sets the gitlab flag when automerge is used too.

How would you communicate this?

Comment (viceice) [reply docs/usage/configuration-options.md]: 🤔

Maybe something like:

If you are using automerge you can use this setting to use the gitlab native automerge function for MR's to speedup automerge.

Comment (kdev) [reply docs/usage/configuration-options.md]: I've updated the file

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -676,6 +676,11 @@ Example:
 
 ## gitLabAutomerge
 
+If you are using automerge you can use this setting to use the gitlab native automerge function for MR's to speedup automerge.
+Attention: This is a top-level key.
+To activate automerging within a `packageRule`, you must set `automerge` to `true` in the same `packageRule`.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:681]:
```suggestion
If you enabled automerge in the Renovate config, you can speed up the automerge process, by using GitLab's own automerge function.
Attention: This is a top-level key.
To activate automerging within a `packageRule`, you must set `automerge` to `true` in the same `packageRule`.
```

The proper noun GitLab should be capitalized like in my suggestion.
Also maybe the new text is clearer/easier to understand?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -676,6 +676,11 @@ Example:
 
 ## gitLabAutomerge
 
+If you enabled automerge in the Renovate config, you can speed up the automerge process, by using GitLab's own automerge function.
+Attention: This is a top-level key.
+To activate automerging within a `packageRule`, you must set `automerge` to `true` in the same `packageRule`.
```
Comment (viceice) [inline docs/usage/configuration-options.md:681]:
```suggestion
```

`automerge` can also be used at toplevel, so this is not right.

**Comment by viceice** [comment]
needs lint fix, run `yarn prettier-fix`

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -681,6 +681,9 @@ Example:
 
 ## gitLabAutomerge
 
+If you enabled automerge in the Renovate config, you can speed up the automerge process, by using GitLab's own automerge function.
```
Comment (viceice) [inline docs/usage/configuration-options.md:684]:
```suggestion
If you enabled `automerge` in the Renovate config, you can speed up the automerge process, by using GitLab's own automerge function.
```

so highlight `automerge` as an renovate option

File: docs/usage/configuration-options.md [hunk 6]
```diff
@@ -681,6 +681,9 @@ Example:
...
+Attention: This is a top-level key.
+
```
Comment (viceice) [inline docs/usage/configuration-options.md:686]:
```suggestion
```

I think thats's not true, you can also use `gitLabAutomerge` in a package rule.

Comment (viceice) [reply docs/usage/configuration-options.md:686]: https://github.com/renovatebot/renovate/blob/34ef699b64b5de1c6f9fe70eafbb99422b839081/lib/workers/pr/index.ts#L115-L128

File: docs/usage/configuration-options.md (hunk 5 above)
Comment (rarkins) [reply docs/usage/configuration-options.md:684]: Doesn't Gitlab automerge speed to renovate automerge, not vice versa?

File: docs/usage/configuration-options.md [hunk 7]
```diff
@@ -681,6 +681,7 @@ Example:
 
 ## gitLabAutomerge
 
+If you enabled `automerge` in the Renovate config, you can speed up the automerge process, by using GitLab's own automerge function.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:684]:
```suggestion
If you enabled `automerge` in the Renovate config, you can speed up the automerge process by using GitLab's own automerge function.
```

I think we can drop the comma here.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**